src/Symfony/Component/Security/Http/Authenticator/AbstractAuthenticator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator;
+
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\PassportInterface;
+use Symfony\Component\Security\Http\Authenticator\Token\PostAuthenticationToken;
+
+/**
+ * An optional base class that creates the necessary tokens for you.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ *
+ * @experimental in 5.1
+ */
+abstract class AbstractAuthenticator implements AuthenticatorInterface
+{
+    /**
+     * Shortcut to create a PostAuthenticationToken for you, if you don't really
+     * care about which authenticated token you're using.
+     *
+     * @return PostAuthenticationToken
+     */
+    public function createAuthenticatedToken(PassportInterface $passport, string $providerKey): TokenInterface
+    {
+        return new PostAuthenticationToken($passport->getUser(), $providerKey, $passport->getUser()->getRoles());
```

Comment:
There is no `PassportInterface::getUser()`. I guess we would need a `$passport instanceof UserPassportInterface` here. If it doesn't implement it, I would probably throw an exception and say "Implement this method yourself!".